### ISSUE TITLE
Create Jenkins pipeline to automate building/testing a test range

### ISSUE THREAD

**Issue by rmfirth** [dev4vater/vater#167]
This is a sub-problem of #140 . We need a Jenkins job that can automatically build a test range, run tests on that range, then destroy the range.

With some enhancements to the vater CLI, this would be fairly straightforward. At a high level, Jenkins needs to be able to invoke a script that can create a project, environment, and inventory that would serve as the context for a "Build Test Range" task and "Destroy Test Range" task. The jenkins pipeline would execute as follows:
1. Use vater to execute a Semaphore task that creates the test project and range
2. Poll the API for task completion
3. When the range has been created, run tests on the range (tests will be implemented as a separate issue)
5. When tests are complete, Use vater to execute a Semaphore task that destroys the range and project

**Comment by rmfirth** [comment]
@uwardlaw @rylagek This pipeline depends heavily on the ability to create a range and project from one Semaphore task. What I had in mind was something akin to the "Create Class" task template, except it would be a "Create Test Class" task template or something, which would create a Semaphore project associated with the Test and the range itself.

My thought is that we need a separate task template because "Create Class" (I am assuming) will be designed to execute in the context of the production environment (i.e. a Semaphore environment with variable values associated with production). This "Create Test Class" task could execute with a different Semaphore environment for staging purposes, but use the same ansible task in the back end (just with different environment variables).

We haven't really discussed in depth how this build/test of the test range would work, and I would like to make sure we are all on the same page. 
Can you validate the following assumptions?
1. Ansible tasks that cause VMs to be built will _eventually_ have things like the vSphere URI passed in as a variable from a Semaphore environment
2. The "Create Class" Semaphore task will have _production_ values (e.g. IP address of Control, IP address of vCenter server, etc.)

Now, assuming those are true, the changes I am proposing from the vater perspective are as follows (please comment with whether or not you agree that this is what needs to happen):
- Implement the "Run Semaphore Task" CLI option

**Comment by uwardlaw** [comment]
1. I'm not sure there's an issue open, but assuming Vault isn't implemented, there will likely be something inserted in `group_vars` see the Terraform [design](https://github.com/uwardlaw/vater/wiki/Terraform) in step 0 for an idea of how this would work.
2. Yes. At this time, the IP address is discovered in `config.py`:

```
 50         ps = subprocess.Popen(('hostname', '-I'), stdout=subprocess.PIPE)
 51         output = subprocess.check_output(('cut', '-d', ' ', '-f1'), stdin=ps.stdout)
 52         cfg['host']['ip'] = (output.decode('utf-8').strip())
```

The Control IP is added to the `Management` project environment, and then `Create Class` calls `rous/bin/setupNewClass.py` with the IP as an argument.

The issue for the Run Semaphore Task is documented in:
- https://github.com/uwardlaw/vater/issues/16

I closed it without implementing `task` for some reason.....shame on me. I'll finish it.

**Comment by uwardlaw** [comment]
Also worth considering is if we rely on Semaphore for the tasks, if Semaphore fails, the rest of the tests can't run. While not the end of the world, I don't really like the idea of not being able to test the components because one piece failed. This would probably cost the team time. For instance, if Semaphore fails, other developers can't start patching issues in the range because until Semaphore is fixed because Semaphore failing prevented the range from being tested.

**Comment by rmfirth** [comment]
@uwardlaw Can you please clarify what you mean by "if Semaphore fails"? One of the reasons it makes sense to use Semaphore here is we are testing an "end-to-end" scenario for our users. In other words, our users are going to be using the Semaphore UI to kick off tasks (like creating ranges, etc.), so we want to test the product the same way. It's definitely an option to look at alternate ways of kicking off the tasks without using Semaphore, but then we are testing a different "end-to-end" flow than what our users will be doing. For our purposes, I feel that our end-to-end test methodology should mirror the user experience as closely as we reasonably can.

**Comment by uwardlaw** [comment]
I don't have much to add to what I already said. If we use Semaphore to test everything else, and Semaphore fails, you can't complete the other tests until Semaphore is fixed.

End to end is great, matching the UE is great, but a single error in Semaphore probably shouldn't stop us from testing a whole range.

Food for thought, not a block concern.

**Comment by marissaeinhorn** [comment]
Not required for RANS implementation or interim solution.  Detailed request for an automated testing method for Foundry content development in #236
